app/views/posts/_form.html.erb

```diff
@@ -0,0 +1,41 @@
+<%= form_with(model: post, local: true) do |form| %>
+  <% if post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% post.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field ">
+    <%= form.label :title %>
+    <%= form.text_field :title, class:"form-control" %>
+  </div>
+
+  <div class="field ">
+    <%= form.label :body %>
+    <%= form.text_area :body, class:"form-control"%>
+  </div>
+
+  <div class="field">
+    <%= form.label :author %>
+    <%= form.text_field :author, class:"form-control" %>
+  </div>
+
+   <div class="field">
+    <% form.label :user_id %>
+    <%= form.number_field :user_id, id: :post_user_id, value: current_user.id, type: :hidden %>
+  </div>
```

Comment:
Please remove this hidden input field. Any logged-in user can alter the `value` attribute and pose as a different author. This is a huge security flaw. 

I have already discussed a solution above. Please follow that and remove this input field.